ledger/participant-integration-api/src/main/scala/platform/store/backend/common/EventStorageBackendTemplate.scala

```diff
@@ -795,17 +795,28 @@ abstract class EventStorageBackendTemplate(
   }
 
   override def maxEventSequentialIdOfAnObservableEvent(
-      offset: Offset
-  )(connection: Connection): Option[Long] = {
+      offsetInclusive: Offset
+  )(connection: Connection): Long = {
+    val ledgerEnd = ledgerEndCache()
     import com.daml.platform.store.backend.Conversions.OffsetToStatement
     SQL"""
-         SELECT
-            event_sequential_id_last
-         FROM
-            participant_transaction_meta
-         WHERE
-            event_offset = (SELECT MAX(event_offset) FROM participant_transaction_meta WHERE event_offset <= $offset)
-       """.as(get[Long](1).singleOpt)(connection)
+     SELECT
+        event_sequential_id_first
+     FROM
+        participant_transaction_meta
+     WHERE
+        event_offset > $offsetInclusive
+        AND event_offset <= ${ledgerEnd._1}
+     ORDER BY
+        event_offset
+     ${QueryStrategy.limitClause(Some(1))}
+   """.as(get[Long](1).singleOpt)(connection)
+      .getOrElse(
+        // after the offset there is no meta, so no tx,
+        // therefore the next (minimum) event sequential id will be
+        // the first event sequential id after the ledger end
+        ledgerEnd._2 + 1
+      ) - 1
```

Comment:
looking up the max of <= offset seq id is equivalent to look up min of > offset seq id minus one, hence the properties of the meta ranges: they are continuous